docs/source/traits_user_manual/debugging.rst

```diff
@@ -4,44 +4,70 @@
 Tips for debugging Traits
 =========================
 
+.. _debugging-change-handler-error:
 
 Re-raising exceptions in change handlers
 ========================================
 
 Traits will typically log (instead of raise) exceptions when an exception is
-encountered in a trait-change handler. This behavior is often preferred in
-applications, since you usually want to avoid critical failures in
-applications. However, when debugging these errors, the
-``logging.Logger.exception`` only displays the tip of the traceback. For example,
-the following code changes a ``constant``:
+encountered in a trait-change handler (see :ref:`observe-notification`). This
+behavior is often preferred in applications, since you usually want to avoid
+critical failures in applications. However, when debugging these errors, the
+``logging.Logger.exception`` only displays the tip of the traceback. For
+example, the following change handler raises an exception:
 
 .. code-block:: python
 
-   from traits.api import HasTraits, Int
+   from traits.api import HasTraits, Int, observe
 
    class Curmudgeon(HasTraits):
-       constant = Int(1)
-       def _constant_changed(self):
+       number = Int(1)
+
+       @observe("number")
+       def number_updated(self):
            raise ValueError()
 
    c = Curmudgeon()
-   c.constant = 42
+   c.number = 42
```

Comment:
I changed the example context not to use a trait named "constant". It suggests one can use change handler for validation, which should be discouraged I think (I used to think that way too).  Took the opportunity to modernize the example to use observe as well.